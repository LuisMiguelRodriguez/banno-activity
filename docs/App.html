<!DOCTYPE html>
<html lang="en">
<head>
    
    <meta charset="utf-8">
    <title>App - Documentation</title>
    
    
    <script src="scripts/prettify/prettify.js"></script>
    <script src="scripts/prettify/lang-css.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc.css">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body>

<input type="checkbox" id="nav-trigger" class="nav-trigger" />
<label for="nav-trigger" class="navicon-button x">
  <div class="navicon"></div>
</label>

<label for="nav-trigger" class="overlay"></label>

<nav >
    
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="App.html">App</a><ul class='methods'><li data-type='method'><a href="App.html#.-MakesanAxiosrequesttoretreivethehtmlfromthespecifiedlink.">- Makes an Axios request to retreive the html from the specified link.</a></li><li data-type='method'><a href="App.html#.componentDidMount">componentDidMount</a></li><li data-type='method'><a href="App.html#.handleAlphaNumerics">handleAlphaNumerics</a></li><li data-type='method'><a href="App.html#.handleFinancialCount">handleFinancialCount</a></li><li data-type='method'><a href="App.html#.handlePngCount">handlePngCount</a></li><li data-type='method'><a href="App.html#.handleProductCount">handleProductCount</a></li><li data-type='method'><a href="App.html#.handleTwitterHandle">handleTwitterHandle</a></li></ul></li></ul><h3>Global</h3><ul><li><a href="global.html#Card">Card</a></li></ul>
</nav>

<div id="main">
    
    <h1 class="page-title">App</h1>
    

    




<section>

<header>
    
        <h2>
        App
        </h2>
        
            <div class="class-description"><p>App Component</p></div>
        
    
</header>

<article>
    
        <div class="container-overview">
        
            

    
    <h2>Constructor</h2>
    

    <h4 class="name" id="App"><span class="type-signature"></span>new App<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line11">line 11</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>
































        
        </div>
    

    

    

    
    
    

     

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id=".state"><span class="type-signature">(static) </span>state<span class="type-signature"></span></h4>





<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line25">line 25</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>



    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>link</code></td>
            

            <td class="type">
            
                
<span class="param-type">String</span>


            
            </td>

            

            

            <td class="description last"><p>Link to be used for webscraping.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>data</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Holds html from dom to be searched through.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>alphaNumerics</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array</span>


            
            </td>

            

            

            <td class="description last"><p>An array that will hold the top 3 alphnumerics found.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>pngCount</code></td>
            

            <td class="type">
            
                
<span class="param-type">Number</span>


            
            </td>

            

            

            <td class="description last"><p>A Count of PNG image found on the page.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>financialIntCount</code></td>
            

            <td class="type">
            
                
<span class="param-type">Number</span>


            
            </td>

            

            

            <td class="description last"><p>A count of how many times financial institute shows on the page.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>twitterHandle</code></td>
            

            <td class="type">
            
                
<span class="param-type">String</span>


            
            </td>

            

            

            <td class="description last"><p>Twitter Handle .</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>productCount</code></td>
            

            <td class="type">
            
                
<span class="param-type">Number</span>


            
            </td>

            

            

            <td class="description last"><p>A count of products found on the page.</p></td>
        </tr>

    
    </tbody>
</table>













        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    <h4 class="name" id=".-MakesanAxiosrequesttoretreivethehtmlfromthespecifiedlink."><span class="type-signature">(static) </span>- Makes an Axios request to retreive the html from the specified link.<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line35">line 35</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>
































        
            

    

    <h4 class="name" id=".componentDidMount"><span class="type-signature">(static) </span>componentDidMount<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line52">line 52</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <ul>
<li>On initial load componentDidMout will call the method getDom() to retreive the html from the link specified.</li>
</ul>
</div>




























        
            

    

    <h4 class="name" id=".handleAlphaNumerics"><span class="type-signature">(static) </span>handleAlphaNumerics<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line63">line 63</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <ul>
<li>This method does a search for all alphanumerics on the page using a regex expression.</li>
</ul>
</div>




























        
            

    

    <h4 class="name" id=".handleFinancialCount"><span class="type-signature">(static) </span>handleFinancialCount<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line127">line 127</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <ul>
<li>Searches page for all occurences of <code>financial institution</code> with a regex expression and keeps a count.</li>
</ul>
</div>




























        
            

    

    <h4 class="name" id=".handlePngCount"><span class="type-signature">(static) </span>handlePngCount<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line108">line 108</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <ul>
<li>This method does a search for all .png occurences on the pages and keeps a count.</li>
</ul>
</div>




























        
            

    

    <h4 class="name" id=".handleProductCount"><span class="type-signature">(static) </span>handleProductCount<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line144">line 144</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <ul>
<li>Searches for products on page by looking under the a dropdown menu which holds all a list of products.</li>
</ul>
</div>




























        
            

    

    <h4 class="name" id=".handleTwitterHandle"><span class="type-signature">(static) </span>handleTwitterHandle<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line170">line 170</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <ul>
<li>Locates the twitter handle in the footer by looking through the li tags.</li>
</ul>
</div>




























        
    

    

    
</article>

</section>




    
    
</div>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sat Mar 23 2019 12:09:04 GMT-0500 (CDT) using the <a href="https://github.com/clenemt/docdash">docdash</a> theme.
</footer>

<script>prettyPrint();</script>
<script src="scripts/linenumber.js"></script>


</body>
</html>