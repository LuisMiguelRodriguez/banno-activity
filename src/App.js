import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import cheerio from 'cheerio';
import axios from 'axios'

class App extends Component {

  state = {
    link: ''
  }

  clickHandler = () => {

    axios('http://www.banno.com')
      .then(({ data }) => {

        const $ = cheerio.load(data);
        const dom = $('body').html();

        // look only for alphanumeric 
        const res = dom.match(/([a-zA-z]+[\d]+)|([\d]+[a-zA-z]+)/g);
        const obj = {};

        // count repetitions of alphanumerics
        res.map(item => {

          if (obj[item] >= 0) {
            obj[item] = obj[item] + 1;

          } else {
            obj[item] = 0;
          }

        })

        // new array to hold alphanumerics count
        let newArr = [];

        for(let temp in obj){
          newArr.push({alpha: temp , count: obj[temp]})
        }

        // Organized array by count value 
        var sortedArr = newArr.sort((current, next) =>  next.count - current.count);
        console.log('== sorted array ===========')

        // Log top 3 highest count
        console.log('#1 ', sortedArr[0])
        console.log('#2 ', sortedArr[1])
        console.log('#3 ', sortedArr[3])


        // Array to hold products found on the page
        const products = [];

        $('#sub-menu').children('li').each((i, item) => {

          products.push($(item).find('span').text().trim());

        })

        console.log('Current Products Listed ', products)

      })
      .catch(err => console.log(err))


  }


  render() {
    return (
      <div className="App">
        <Button onClick={this.clickHandler} letiant="primary" size="lg" block>
          Scan
        </Button>
      </div>
    );
  }
}

export default App;
