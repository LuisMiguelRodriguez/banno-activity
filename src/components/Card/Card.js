import React from 'react';
import { Card, Button } from 'react-bootstrap';

export default props => {
    
    return (
        <Card className="text-center">
            <Card.Body>
                <Card.Title>Activity - {props.activity}</Card.Title>
                <Button onClick={props.onClickHandler} variant="primary">Run Activity</Button>
            </Card.Body>
            <Card.Footer className="">{props.result ? props.result : ''}</Card.Footer>
            {props.array ? props.array.map((item, i )=> {
                return <Card.Footer className="" key={i}>{`AlphaNumeric: ${item.alpha}  Count: ${item.count}`}</Card.Footer>
            }) : <h1></h1>}
        </Card>
    )
}
