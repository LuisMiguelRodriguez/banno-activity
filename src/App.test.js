import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
import { shallow } from 'enzyme';
import App from './App';
import Card from './components/Card';

const app = shallow(<App />);

describe('App', () => {

    describe('renders correctly', () => {

        it('it looks the same as snapshot', () => {
            expect(app).toMatchSnapshot();
        })
    });

    describe('renders five <Card/> components ', () => {
        expect(app.find(Card)).toHaveLength(5)
    })

    describe('Checking Initial State', () => {


        it('initializes the `state` with an empty list of alphaNumerics', () => {
            expect(app.state().alphaNumerics).toEqual([]);
        })

        it('initializes the `state` with an empty string for twitterHandle ', () => {
            expect(app.state().twitterHandle).toEqual('');
        })

        it('initializes the `state` with an empty string for link ', () => {
            expect(app.state().link).toEqual('');
        })

        it('initializes the `state` with  0 for pngCount', () => {
            expect(app.state().pngCount).toEqual(0);
        })


        it('initializes the `state` with  0 for productCount', () => {
            expect(app.state().productCount).toEqual(0);
        })
    })



    // describe('when clicking the `add-gift` button', () => {

    //     const id = 1;

    //     beforeEach(() => {
    //         app.find('.btn-add').simulate('click');
    //     });

    //     afterEach(() => {
    //         app.setState({ gifts: [] })
    //     });
    //     it('adds a new gift to `state` ', () => {
    //         expect(app.state().gifts).toEqual([{ id: 1 }])
    //     });

    //     it('adds a new gift to the rendered list ', () => {
    //         expect(app.find('.gift-list').children().length).toEqual(1);
    //     });

    //     it('Creates a Gift component', () => {
    //         expect(app.find('Gift').exists()).toBe(true);
    //     });

    //     describe('and the user wants to remove the added gift', () => {
    //         beforeEach(()=> {
    //             app.instance().removeGift(id);
    //         });

    //         it('removes the gift from `state`', () => {
    //             expect(app.state().gifts).toEqual([]);
    //         });
    //     });

    // });


})
