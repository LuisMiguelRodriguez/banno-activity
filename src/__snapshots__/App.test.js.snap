// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly it looks the same as snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ForwardRef(Bootstrap(Container))>
        <ForwardRef(Bootstrap(Jumbotron))
          style={
            Object {
              "marginTop": "20px",
              "textAlign": "center",
            }
          }
        >
          <h1>
            Banno Assignment
          </h1>
        </ForwardRef(Bootstrap(Jumbotron))>
        <CardC
          activity="Alpha Numeric Top 3"
          array={Array []}
          onClickHandler={[Function]}
        />
        <CardC
          activity="PNG Count"
          onClickHandler={[Function]}
        />
        <CardC
          activity="Count of Financial Instituation "
          onClickHandler={[Function]}
          result={0}
        />
        <CardC
          activity="Count of Products"
          onClickHandler={[Function]}
          result={0}
        />
        <CardC
          activity="Locate Twittle Handle Dynamically"
          onClickHandler={[Function]}
          result=""
        />
      </ForwardRef(Bootstrap(Container))>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef(Bootstrap(Jumbotron))
            style={
              Object {
                "marginTop": "20px",
                "textAlign": "center",
              }
            }
          >
            <h1>
              Banno Assignment
            </h1>
          </ForwardRef(Bootstrap(Jumbotron))>,
          <CardC
            activity="Alpha Numeric Top 3"
            array={Array []}
            onClickHandler={[Function]}
          />,
          <CardC
            activity="PNG Count"
            onClickHandler={[Function]}
          />,
          <CardC
            activity="Count of Financial Instituation "
            onClickHandler={[Function]}
            result={0}
          />,
          <CardC
            activity="Count of Products"
            onClickHandler={[Function]}
            result={0}
          />,
          <CardC
            activity="Locate Twittle Handle Dynamically"
            onClickHandler={[Function]}
            result=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h1>
              Banno Assignment
            </h1>,
            "style": Object {
              "marginTop": "20px",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Banno Assignment",
            },
            "ref": null,
            "rendered": "Banno Assignment",
            "type": "h1",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activity": "Alpha Numeric Top 3",
            "array": Array [],
            "onClickHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activity": "PNG Count",
            "onClickHandler": [Function],
            "result": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activity": "Count of Financial Instituation ",
            "onClickHandler": [Function],
            "result": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activity": "Count of Products",
            "onClickHandler": [Function],
            "result": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activity": "Locate Twittle Handle Dynamically",
            "onClickHandler": [Function],
            "result": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef(Bootstrap(Container))>
          <ForwardRef(Bootstrap(Jumbotron))
            style={
              Object {
                "marginTop": "20px",
                "textAlign": "center",
              }
            }
          >
            <h1>
              Banno Assignment
            </h1>
          </ForwardRef(Bootstrap(Jumbotron))>
          <CardC
            activity="Alpha Numeric Top 3"
            array={Array []}
            onClickHandler={[Function]}
          />
          <CardC
            activity="PNG Count"
            onClickHandler={[Function]}
          />
          <CardC
            activity="Count of Financial Instituation "
            onClickHandler={[Function]}
            result={0}
          />
          <CardC
            activity="Count of Products"
            onClickHandler={[Function]}
            result={0}
          />
          <CardC
            activity="Locate Twittle Handle Dynamically"
            onClickHandler={[Function]}
            result=""
          />
        </ForwardRef(Bootstrap(Container))>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(Bootstrap(Jumbotron))
              style={
                Object {
                  "marginTop": "20px",
                  "textAlign": "center",
                }
              }
            >
              <h1>
                Banno Assignment
              </h1>
            </ForwardRef(Bootstrap(Jumbotron))>,
            <CardC
              activity="Alpha Numeric Top 3"
              array={Array []}
              onClickHandler={[Function]}
            />,
            <CardC
              activity="PNG Count"
              onClickHandler={[Function]}
            />,
            <CardC
              activity="Count of Financial Instituation "
              onClickHandler={[Function]}
              result={0}
            />,
            <CardC
              activity="Count of Products"
              onClickHandler={[Function]}
              result={0}
            />,
            <CardC
              activity="Locate Twittle Handle Dynamically"
              onClickHandler={[Function]}
              result=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h1>
                Banno Assignment
              </h1>,
              "style": Object {
                "marginTop": "20px",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Banno Assignment",
              },
              "ref": null,
              "rendered": "Banno Assignment",
              "type": "h1",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activity": "Alpha Numeric Top 3",
              "array": Array [],
              "onClickHandler": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activity": "PNG Count",
              "onClickHandler": [Function],
              "result": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activity": "Count of Financial Instituation ",
              "onClickHandler": [Function],
              "result": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activity": "Count of Products",
              "onClickHandler": [Function],
              "result": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activity": "Locate Twittle Handle Dynamically",
              "onClickHandler": [Function],
              "result": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
